src/facade/facade_types.h

```diff
@@ -56,6 +60,23 @@ struct ConnectionStats {
   ConnectionStats& operator+=(const ConnectionStats& o);
 };
 
+struct ErrorReply {
+  explicit ErrorReply(std::string msg, std::string_view kind = {})
+      : message{move(msg)}, kind{kind} {
+  }
+  explicit ErrorReply(std::string_view msg, std::string_view kind = {}) : message{msg}, kind{kind} {
+  }
+  explicit ErrorReply(const char* msg, std::string_view kind = {})
+      : message{std::string_view{msg}}, kind{kind} {
+  }
+  explicit ErrorReply(OpStatus status) : message{}, kind{}, status{status} {
+  }
+
+  std::variant<std::string, std::string_view> message;
```

Comment:
Maybe we can have `message` of type `string_view`, and some other member (like private `message_memory_`) which `message` may or may not point into?
It should make using this struct easier

Reply:
I remember you don't like using variants for storage at all, we already had this case once.
But I like to avoid self referential structs - it requires defining move/copy constructors and operators

Reply:
I am not worried about performance implications of using std::string in case we return errors.

Reply:
This is why I use string_view whenever its possible, in other cases you construct a string with StrCat and others anyways

I'll check once again for performance degradation, but it shouldn't be the case

Reply:
Btw I once thought of supporting an `absl::Span<absl::Alphanumeric>` interface to write directly into the conn buffer without the intermediary string, but its just a random comment in this context 🙂

Reply:
So yes, I don't see any place where the number of intermediaries increased

Reply:
I meant - I am NOT worried - i.e. I think having a string here is perfectly fine since it's not on a hot path.

Reply:
I don't have any problem with storing `variant` as a member, or using it as storage. I actually think `variant`s are great.
I dislike the API that it brings in, though. I definitely understand your comment re/ having to implement an explicit move / copy operations.
I guess we could:
1. Keep as is
2. Provide a wrapper get-method that returns `string_view` which points to the right string
3. do my trick, but have the `string` inside a `unique_ptr`
4. always use a `string` (Roman's suggestion)
Your call :)

Reply:
I see you decided to keep as is. I would still say it's not a great API to force a call to `std::visit` with a templated lambda instead of having a
```cpp
string_view GetMessage() const { visit... }
```
To wrap that, but again, your call :)

For the case where you want to move if possible, you could have another version (`GetMessageDestructive()`?)

Reply:
😝 
As soon as it'll be used more than in once place I'll add those